internal/network/data_sender.go

```diff
@@ -48,6 +48,9 @@ type DataSender interface {
 	SendDeviceStatusData(data any) error
 	// SendDeviceNetworkData sends queried network data over mqtt to a separate network topic
 	SendDeviceNetworkData(data models.DeviceNetworkData) error
+	// SetVehicleInfo sets the vehicle info for the data sender
+	// todo we need somehow to set VehicleInfo differently, we need better separation of concerns here
```

Comment:
I'm guessing this is b/c there could be situations where we are unable to get the VehicleInfo where we still want to send logs, might be fine

Reply:
yes, exactly